README.md

```diff
@@ -556,6 +556,94 @@ should container an array of identifiers. If an identifier is in the array then
     />
 ```
 
+### Externally manage checkboxes
+
+Combining `isCheckboxChecked`, `onMasterCheckboxChange` and `onCheckboxChange` allows a row's checkbox state to be managed outside of the datatable while still allowing `disabledCheckboxes` to work as intended.
+
+```jsx
+const checked = new Set
+
+<DynamicDataTable
+    isCheckboxChecked={({ id }) => checked.has(id)}
+    onMasterCheckboxChange={(_, rows) => {
+        let all = true
+
+        rows.forEach(({ id }) => checked.has(id) || all = false)
+
+        rows.forEach(({ id }) => {
+            if (all) {
+                checked.delete(id)
+            } else if (!checked.has(id)) {
+                checked.add(id)
+            }
+        })
+    }}
+    onCheckboxChange={(_, { id }) => {
+        if (checked.has(id)) {
+            checked.delete(id)
+        } else {
+            checked.add(id)
+        }
+    }}
+/>
+```
+
+#### `isCheckboxChecked`
+
+`isCheckboxChecked` is called on each re-render of the datatable allowing for custom logic to determine if a checkbox is checked. It will recieve the current row and the visibile rows as arguments.
+
+```jsx
+const checked = new Set
+
+<DynamicDataTable
+    isCheckboxChecked={({ id }) => checked.has(id)}
+/>
+```
+
+_Note: This should only be used if `onMasterCheckboxChange` and `onCheckboxChange` are implemented_
+
+#### `onMasterCheckboxChange`
+
+`onMasterCheckboxChange` is called when the master checkbox is clicked. This allows for custom logic for selecting and deselecting multiple rows. It will recieve an event object from the input and the visible rows.
```

Comment:
```suggestion
`onMasterCheckboxChange` is called when the master checkbox is clicked. This allows for custom logic for selecting and deselecting multiple rows. It will receive an event object from the input and the visible rows.

The master checkbox refers to the bulk select checkbox found in the top left corner of the data table whenever checkboxes are enabled.
```